daemon/jobs/mempoolsync.go

```diff
@@ -26,6 +29,11 @@ var mempoolBlock *model.Block
 func MempoolSync() {
 	if !mempoolSyncIsRunning {
 		mempoolSyncIsRunning = true
+		// Need to lock block processing to avoid race condition where we are saving a mempool transaction after a block
+		// has already started processing transactions. The mempool transaction could overwrite the block transaction
+		// incorrectly.
+		processing.BlockLock.Lock()
+		defer processing.BlockLock.Unlock()
```

Comment:
do you need to lock the whole function? can you have the lock more localized?

Reply:
I was doing it at a the tx level, but I had to move it higher up to here because I grab all the current mempool transactions now in the db and I need to lock block processing when I do this to avoid a block saving updates to the tx and then me deleting the transaction. This would cause a bunch of issues.

Reply:
I was doing it at a the tx level, but I had to move it higher up to here because I grab all the current mempool transactions now in the db and I need to lock block processing when I do this to avoid a block saving updates to the tx and then me deleting the transaction. This would cause a bunch of issues.